cmd/cluster/cluster_create.go

```diff
@@ -0,0 +1,252 @@
+		"automatically deleted within two hours. Only the CD-as-a-Service sample application (potato-facts) and remote network agent can be installed."
+)
+
+var (
+	ErrOutputTypeNotSupported = errors.New("output type is not supported. Choose type 'text' to use this feature")
+	ErrWritingSandboxSaveData = errors.New("unable to save sandbox data to file system")
+)
+
+type CreateOptions struct {
+	Context       context.Context
+	ArmoryClient  *configuration.ConfigClient
+	configuration *config.Configuration
+	contextNames  []string
+	credentials   *model.Credential
+	progressbar   *progressbar.ProgressBar
+	saveData      *model.SandboxClusterSaveData
+}
+
+func NewCreateClusterCmd(configuration *config.Configuration) *cobra.Command {
+	o := NewCreateOptions()
+
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{},
+		Short:   createClusterShort,
+		Long:    createClusterLong,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			o.InitializeConfiguration(configuration)
+			if err := o.Run(cmd); err != nil {
+				return err
+			}
+			return nil
+		},
+		SilenceUsage: true,
+	}
+	return cmd
+}
+
+func NewCreateOptions() *CreateOptions {
+	return &CreateOptions{}
+}
+
+// InitializeConfiguration will fetch Auth info to configure the REST client, so initialization must be deferred to runtime
+func (o *CreateOptions) InitializeConfiguration(cfg *config.Configuration) {
+	o.configuration = cfg
+	ac := configuration.NewClient(cfg)
+	o.ArmoryClient = ac
+	o.Context = o.ArmoryClient.ArmoryCloudClient.Context
+}
+
+// Run performs the execution of 'cluster create' sub command and saves the info for later use
+func (o *CreateOptions) Run(cmd *cobra.Command) error {
+	if o.configuration.GetOutputType() != output.Text {
+		return ErrOutputTypeNotSupported
+	}
+	ctx := cmd.Context()
+	agentPrefix := randomString(6)
+	credentials, err := o.ArmoryClient.Credentials().Create(ctx, o.createNamedCredential(agentPrefix))
+	if err != nil {
+		return err
+	}
+	environmentId := lo.If(lo.FromPtrOr[bool](o.configuration.GetIsTest(), false), "test-env").ElseF(o.configuration.GetCustomerEnvironmentId)
+	err = AssignCredentialRNARole(ctx, credentials, o.ArmoryClient, environmentId)
+	if err != nil {
+		return err
+	}
+	createSandboxRequest := o.createSandboxRequest(agentPrefix, credentials)
+	sandboxResponse, err := o.ArmoryClient.Sandbox().Create(ctx, createSandboxRequest)
+	if err != nil {
+		return err
+	}
+	o.InitializeProgressBar()
+	o.saveData = &model.SandboxClusterSaveData{
+		SandboxCluster:        model.SandboxCluster{},
+		CreateSandboxRequest:  *createSandboxRequest,
+		CreateSandboxResponse: *sandboxResponse,
+	}
+	for {
+		cluster, err := o.ArmoryClient.Sandbox().Get(ctx, o.saveData.CreateSandboxResponse.ClusterId)
+		if err != nil {
+			return err
+		}
+
+		done, err := o.UpdateProgressBar(cluster)
+		if err != nil {
+			return err
+		}
+
+		if done {
+			break
+		}
+	}
```

Comment:
Both comments are completely fair. It definitely makes sense to tell them what to do next! I commented above that I was storing metadata about the cluster to make it easier to use it via the CLI later (in future commands). I was honestly unsure if we'd want to inform them about the agent automatically installed because in the cdaas-examples onboarding, we'll ask them to install an agent. We could say "skip this if you use a temp cluster", but maybe we just want them to do that anyway and not worry about what the temp cluster did. I'd like to discuss it offline